AMX gitbash assignment submission

asutosh29 opened the pull request:
## Assignment parts
1. git fracz writeup
2. bandits writeup 
3. site tracker script

asutosh29 committed:
create amx directory for submission

asutosh29 committed:
Solve git assignment till ignore-them

asutosh29 committed:
Solved till merge-conflict

asutosh29 committed:
Solve till save-your-work

asutosh29 committed:
Solve till fix-typo

asutosh29 committed:
Solve till fix-old-typo

asutosh29 committed:
Solve bandits till Level 4

asutosh29 committed:
Solve bandits till level 15

asutosh29 committed:
Create pinger script

asutosh29 committed:
Setup getopts to take options and flags from user

asutosh29 committed:
Add function to add site to tracklist. Add check if tracklist file exists

asutosh29 committed:
Add function to delete site from tracklist

asutosh29 committed:
Add help function which shows instruction for user

asutosh29 committed:
Add pinger function to get site status

asutosh29 committed:
Add list site function to view all sites in tracklist

asutosh29 committed:
Edit pinger help menu. Splits Options and Flags

asutosh29 committed:
SOlve bandits till level 19

asutosh29 committed:
Add More instructions in help page. Hide tracklist file.

ayu-ch commented:
```
$ ./pinger.sh  -l
google.com

$ ./pinger.sh -p
Pinging google.com
Site is down with status code: 
```

This is the output I'm getting

ayu-ch commented on amx/pinger.sh:
```diff
@@ -0,0 +1,115 @@
+
+# Tool used to moniter status of websites
+
+if ! [ -f ./.tracklist.txt ]
```
`.tracklist.txt`?

asutosh29 replied:
This file stores all the names of the sites to be tracked. I run a check to make sure that the file exists before proceeding with other functions

ayu-ch replied:
but that file is tracklist.txt right?

ayu-ch replied:
what is the difference between .tracklist.txt and tracklist.txt?

ayu-ch replied:
oh okay, you removed it now

ayu-ch commented on amx/pinger.sh:
```diff
@@ -0,0 +1,115 @@
+then
+    # echo "Tracklist doesnt exists"
+    # echo "Creating tracklist.txt"
+    touch ./.tracklist.txt
```
here as well

asutosh29 commented:
> ```
> $ ./pinger.sh  -l
> google.com
> 
> $ ./pinger.sh -p
> Pinging google.com
> Site is down with status code: 
> ```
> 
> This is the output I'm getting

input should be www.google.com i have added this in help page also as an example.

asutosh29 commented:
> > ```
> > $ ./pinger.sh  -l
> > google.com
> > 
> > $ ./pinger.sh -p
> > Pinging google.com
> > Site is down with status code: 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is the output I'm getting
> 
> input should be [www.google.com](http://www.google.com) i have added this in help page also as an example.

Fixed that. It was a redirect problem. Added -L flag while making the curl request

asutosh29 committed:
Add redirect in curl request. Fix empty file error message. Add check for empty file before listing added sites.

asutosh29 committed:
Add index to site while listing

asutosh29 committed:
Add delete site by index feature

asutosh29 committed:
Add edit site by index feature

asutosh29 committed:
Add Banner and Color list

asutosh29 committed:
Add colors to script

asutosh29 committed:
Add site validity checker. Fix: Check for exact site match before adding to list.

asutosh29 committed:
Add minor UI changes
